server/graphql/v1/mutations/orders.js

```diff
@@ -785,19 +785,24 @@ export async function refundTransaction(_, args, req) {
   const collective = await models.Collective.findByPk(transaction.CollectiveId);
   const isHost = await collective.isHost();
   const HostCollectiveId = isHost ? collective.id : await collective.getHostCollectiveId();
+  const timeLimit = moment().subtract(30, 'd');
+  const createdAtMoment = moment(transaction.createdAt);
+  const transactionIsOlderThanThirtyDays = createdAtMoment < timeLimit;
 
-  // 1. Verify user permission. User must be either
-  //   a. Admin of the collective that received the donation - disabled for now
+  // 1a. Verify user permission. User must be either
+  //   a. Admin of the collective that received the donation
   //   b. Admin of the Host Collective that received the donation
   //   c. Admin of opencollective.com/opencollective
 
-  if (
-    !req.remoteUser.isAdmin(HostCollectiveId) &&
-    !req.remoteUser.isRoot() /* && !req.remoteUser.isAdmin(collective.id) */
-  ) {
+  if (!req.remoteUser.isAdmin(HostCollectiveId) && !req.remoteUser.isRoot() && !req.remoteUser.isAdmin(collective.id)) {
```

Comment:
Here we don't check wether the transaction is a CREDIT or DEBIT. It means that, by passing the proper id, contributors could be able to refund their own transactions.

Reply:
Can you clarify what you mean? Do I need to make sure the transaction is only refunded if it isn't CREDIT or DEBIT? Or only if it isn't? Or to make sure that it's one or the other?

Reply:
Looking at `canRefund` in `transactions.ts`, we check whether it's CREDIT or not and if it isn't, we can't refund. So does that take care of your issue? Or do you also want to add that to the line you've highlighted here?

```
export const canRefund = async (transaction, _, req): Promise<boolean> => {
  if (transaction.type !== TransactionTypes.CREDIT || transaction.OrderId === null) {
    return false;
  }
  return remoteUserMeetsOneCondition(req, transaction, [isRoot, isPayeeHostAdmin, isPayeeCollectiveAdmin]);
};
```

Reply:
> Looking at canRefund in transactions.ts, we check whether it's CREDIT or not and if it isn't, we can't refund. So does that take care of your issue?

But unless I'm mistaking `canRefund` is not called from this `refundTransaction`.

The idea is that if I (collective 1) make a contribution to RandomCollective (collective 2), the transactions pair will look like that:
> { id: 1, type: DEBIT, FromCollectiveId: 2, CollectiveID: 1 }
> { id: 2, type: CREDIT, FromCollectiveID: 1, CollectiveId: 2 }

The happy path to refund the transaction is: `refundTransaction(2)`. We'll check that you're an admin of collective 2, and if yes than everything's fine.

But let's say I'm a malicious user, and I call this endpoint as `refundTransaction(1)`. Now the collective ID (1) will be my user profile, so I'll be allowed to refund my own transaction.

Note: I haven't tested this to confirm, it's only theoretical.

Reply:
Ok, thanks for explaining! I understand better now. I didn't know about the DEBIT/CREDIT model of transactions, I thought that meant the card type 🤦‍♀️ I will add that check in `refundTransaction`